src/Databases/DatabasesCommon.cpp

```diff
@@ -149,6 +149,12 @@ ASTPtr getCreateQueryFromStorage(const StoragePtr & storage, const ASTPtr & ast_
                         return nullptr;
                 }
                 ast_column_declaration->type = ast_type;
+
+                if (auto column_default = metadata_ptr->columns.getDefault(column_name_and_type.name))
+                {
+                    ast_column_declaration->default_specifier = toString(column_default->kind);
+                    ast_column_declaration->default_expression = column_default->expression;
+                }
```

Comment:
If a `StorageDistributed` is created without a list of columns and without the `AS` clause it extracts the list of columns from its target storage, see https://github.com/ClickHouse/ClickHouse/blob/e9c3032f838dcd20eda22adf04a425f88b862eaf/src/Storages/StorageDistributed.cpp#L346